sklearn/tests/test_pipeline.py

```diff
@@ -558,15 +558,31 @@ def test_pipeline_fit_transform():
     assert_array_almost_equal(X_trans, X_trans2)
 
 
-def test_pipeline_slice():
-    pipe = Pipeline([('transf1', Transf()),
-                     ('transf2', Transf()),
-                     ('clf', FitParamT())])
-    pipe2 = pipe[:-1]
-    assert isinstance(pipe2, Pipeline)
-    assert pipe2.steps == pipe.steps[:-1]
-    assert 2 == len(pipe2.named_steps)
-    assert_raises(ValueError, lambda: pipe[::-1])
+@pytest.mark.parametrize("start, end", [(0, 1), (0, 2), (1, 2), (1, 3)])
```

Comment:
Please test one or both of the indices being None

Reply:
Done @jnothman